src/common/metrics/ngx_wa_histogram.c

```diff
@@ -96,7 +92,8 @@ histogram_bin(ngx_wa_metrics_t *metrics, ngx_wa_metrics_histogram_t *h,
             return b;
         }
 
-        if (b->upper_bound == 0) {
+        if (b->upper_bound == NGX_MAX_UINT32_VALUE) {
+            i++;
```

Comment:
This change feels like it is something else than exposing histograms to `ngx_wa_metrics_get`, can you elaborate?

Reply:
Indeed, this change is not required to have histograms exposed in `ngx_wa_metrics_get`.

This change is meant to make the code clearer and more closely aligned with the design of histograms' bins -- it doesn't actually change the behaviour of the iteration itself.

Histograms are created with 5 bins, the first one initialized with  `NGX_MAX_UINT32_VALUE` as upper bound:
`[(0, NGX_MAX_UINT32_VALUE), (0, 0), (0, 0), (0, 0), (0, 0)]`.

When a new value, e.g. 1, is added, the first bin is shifted to the right to make room for the new bin:
`[(1, 2), (0, NGX_MAX_UINT32_VALUE), (0, 0), (0, 0), (0, 0)]`.

In this design, the upper bound `NGX_MAX_UINT32_VALUE` always signals the end of the initialized bins -- the bins to the right aren't initialized and shouldn't be visited. 

Previously, the code was iterating the bins until it found one whose `upper_bound` was 0. That worked, but it should really be iterating over the initialized bins only.

Reply:
Ok got it, thanks for the explanation!